cmd/kubeadm/app/util/etcd.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+type etcdVersionResponse struct {
+	Etcdserver  string `json:"etcdserver"`
+	Etcdcluster string `json:"etcdcluster"`
+}
+
+const (
+	internalEtcdRequestTimeout  = time.Duration(10 * time.Second)
+	internalEtcdRequestRetries  = 3
+	internalEtcdRequestInterval = time.Duration(5 * time.Second)
+)
+
+// GetEtcdRunningVersion returns a version of etcd running on a local host
+func GetEtcdRunningVersion() (string, error) {
+	var internalEtcdEndpoint = "http://127.0.0.1:2379"
```

Comment:
nit: Why don't we use `github.com/coreos/etcd/client` or `github.com/coreos/etcd/clientv3` here?
I see the v2 `Client` object might be able to invoke a normal `/version` request against the server.
Then it'd be great to use the v3 `Client` possibly to get the member status as an extra check.

Anyway, that might be overkill for now